alea/src/batch.cpp

```diff
@@ -89,10 +92,27 @@ void batch_acc<T>::add(const computed<T> &source, size_t count)
         next_batch();
 
     // Since Eigen matrix are column-major, we can just pass the pointer
-    source.add_to(sink<T>(store_->batch().col(cursor_.current()).data(), size()));
+    source.add_to(view<T>(store_->batch().col(cursor_.current()).data(), size()));
     store_->count()(cursor_.current()) += count;
 }
 
+template <typename T>
+batch_acc<T> &batch_acc<T>::operator<<(const batch_result<T> &other)
+{
+    internal::check_valid(*this);
+    if (size() != other.size())
+        throw size_mismatch();
+    // TODO this is not strictly speaking necessary when done properly
+    if (num_batches() != other.num_batches())
+        throw size_mismatch();
+
+    // FIXME: this is a terrible idea because it mixes two time series
```

Comment:
Just forbid merging results into `batch_acc` then?

Reply:
> Just forbid merging results into batch_acc then?

Hmmm, I would like to keep it because the mean is still computed correctly, and in principle it is sound to merge to batch accumulators.   The variance will be strange if you have correlated batches then, but the same issue is present with `reduce()` right now.